### ISSUE TITLE
Test dumb-jump-pick-grep-variant-git-grep-in-git-repo failed

### ISSUE THREAD

**Issue by dogsleg** [jacktasia/dumb-jump#194]
Hi,

while running tests I'm getting one test failed.

    Test dumb-jump-pick-grep-variant-git-grep-in-git-repo backtrace:
      (if (unwind-protect (setq value-792 (apply fn-790 args-791)) (setq f
      (let (form-description-794) (if (unwind-protect (setq value-792 (app
      (let ((value-792 (quote ert-form-evaluation-aborted-793))) (let (for
      (let ((fn-790 (function generator-plist-equal)) (args-791 (list gen-
      (let* ((dumb-jump-force-searcher nil) (gen-funcs (dumb-jump-generato
      (closure (t) nil (let* ((dumb-jump-force-searcher nil) (gen-funcs (d
      ert--run-test-internal([cl-struct-ert--test-execution-info [cl-struc
      ert-run-test([cl-struct-ert-test dumb-jump-pick-grep-variant-git-gre
      ert-run-or-rerun-test([cl-struct-ert--stats t [[cl-struct-ert-test d
      ert-run-tests(t #[385 "\306\307\"\203G\211\211G\310U\203\211@\20
      ert-run-tests-batch(nil)
      ert-run-tests-batch-and-exit()
      eval((ert-run-tests-batch-and-exit))
      command-line-1(("-l" "package" "--eval" "(add-to-list 'package-direc
      command-line()
      normal-top-level()
    Test dumb-jump-pick-grep-variant-git-grep-in-git-repo condition:
        (ert-test-failed
         ((should
           (generator-plist-equal gen-funcs variant))
          :form
          (generator-plist-equal
           (:parse dumb-jump-parse-git-grep-response :generate dumb-jump-generate-git-grep-command         :installed dumb-jump-git-grep-installed\? :searcher git-grep)
           (:parse dumb-jump-parse-ag-response :generate dumb-jump-generate-ag-command :installed     dumb-jump-ag-installed\? :searcher ag))
          :value nil))
       FAILED  101/137  dumb-jump-pick-grep-variant-git-grep-in-git-repo

**Comment by dogsleg** [comment]
Hmm, it is a locale-related issue. With LANG=C.UTF-8 LC_ALL=C.UTF-8 prefixed the test passes correctly, but I'm getting another thing:

    ("rg FAILURE '(defvar test ' NOT in response '' | CMD: 'git grep --color=never -h --untracked -E  \\\\\\
    (defvartest\\\\b\\\\s\\*' | rule: '(:type variable :supports (ag grep rg git-grep) :language elisp :regex \\
    (defvarJJJ\\b\\s* :tests ((defvar test ))'")

**Comment by jacktasia** [comment]
Thanks for the report.

> but I'm getting another thing:

Assuming that's happening just before ` passed  139/144  dumb-jump-test-git-grep-rules-fail-test` then that's expected because we're testing an expected failure but the function outputs on errors to be helpful which is admittedly confusing in this case. I'll look into making it not output in this case, but please let me know if that's where you're seeing that message. Thanks!

**Comment by dogsleg** [comment]
I see that message exactly here:

    passed  134/140  dumb-jump-test-ag-rules-test
    ("rg FAILURE '(defvar test ' NOT in response '' | CMD: 'git grep --color=never -h --untracked -E  \\\\\\
    (defvartest\\\\b\\\\s\\*' | rule: '(:type variable :supports (ag grep rg git-grep) :language elisp :regex \\
    (defvarJJJ\\b\\s* :tests ((defvar test ))'")
    passed  135/140  dumb-jump-test-git-grep-rules-fail-test

**Comment by jacktasia** [comment]
Thanks. Looks expected since it's right before the `dumb-jump-test-git-grep-rules-fail-test` I mentioned, but since you're showing a lower amount of total tests (140 vs 144) I have to assume you're running an older version. I'd recommend updating to the most recent version if possible, but tests might fail if you do (which should be fixed with #193), because dependency f.el recently made a backwards incompatible change.

**Comment by dogsleg** [comment]
Thanks for your responses! I'm running 0.5.1 (from Sep 2017).

**Comment by jacktasia** [comment]
> Thanks for your responses! 

No problem!

> I'm running 0.5.1 (from Sep 2017).

Thanks for the info. I really need to release a new stable version since the master branch has definitely changed a lot since the `0.5.1` release. I plan on doing a new release shortly after I merge #193.

**Comment by dogsleg** [comment]
OK, thanks! I'm closing the issue.

**Comment by dogsleg** [comment]
Hi Jack!

Sorry for disturbing and raising this issue. You wrote that you plan to release a new stable version after #193 merging. As I can see it was merged more than a month ago. Could you, please, prepare a new release?

I've packaged [dumb-jump](https://tracker.debian.org/pkg/dumb-jump-el) for Debian and uploaded it to the Debian archive. Unfortunately, due to some (unknown yet) changes an [issue](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=891720) with the package emerged: two tests fail. I've checked current master branch of your repository and as I can see it was fixed. It would be nice to have the fix in Debian together with all the other changes. I wouldn't say that there's any hurry since the next stable Debian release is planned on mid-2019, so please consider my message as a discreet and gentle remainder of your plans. ;-)

**Comment by jacktasia** [comment]
> You wrote that you plan to release a new stable version after #193 merging.

Whoops, yup.  Sorry! Thank you for reminding me.

> As I can see it was merged more than a month ago. Could you, please, prepare a new release?

Yup, setting a reminder to do a release right when I get home from work tonight. I'll update this issue when I do. Thanks again!

**Comment by jacktasia** [comment]
@dogsleg I released [v0.5.2 ](https://github.com/jacktasia/dumb-jump/releases/tag/v0.5.2). Thanks again for the reminder and sorry for the delay. Please let me know if I can help with anything else. Thanks!

**Comment by dogsleg** [comment]
Thank you very much, @jacktasia!
